timemachine/fe/free_energy.py

```diff
@@ -340,14 +351,49 @@ def sample(initial_state: InitialState, md_params: MDParams, max_buffer_frames:
         store_x_interval=0,
     )
 
+    rng = np.random.default_rng(md_params.n_frames * md_params.steps_per_frame)
+
     assert np.all(np.isfinite(ctxt.get_x_t())), "Equilibration resulted in a nan"
 
     def run_production_steps(n_steps: int) -> Tuple[NDArray, NDArray]:
-        _, coords, boxes = ctxt.multiple_steps_U(
-            n_steps=n_steps,
-            store_u_interval=0,
-            store_x_interval=md_params.steps_per_frame,
-        )
+        if md_params.local_steps <= 0:
```

Comment:
Could `md_params.local_steps < 0` just be disallowed by assertion in `MDParams.__post_init__`?

Reply:
Asserted as apart of https://github.com/proteneer/timemachine/pull/1026/commits/9a63ea2b3967cf830b6b599e06dfeddf96b69a45